### ISSUE TITLE
navigateComplete/init event?

### ISSUE THREAD

**Issue by paulwalker** [angular/router#309]
ng-init does not seem to work on elements inside an outlet and navigation is not possible within the controller activate function. it seems as if the workflow is missing one last event?

**Comment by SanderElias** [comment]
@paulwalker, did you notice the canActivate handler?

**Comment by paulwalker** [comment]
yes, but I am looking for something after navigation is complete, not before...nor am i looking to prevent the navigation.

**Comment by SanderElias** [comment]
You are talking about that navigation is not possible, and you want to do it after navigation?
You can inject anything you like into the activate function. What is it you are trying to do?

**Comment by paulwalker** [comment]
rootRouter.navigating evaluates to true during the activate function, preventing navigation (the code just returns an empty resolve). So, it's two things really that are coming out of my experimentation with this. 1) the ability to prevent/change navigation during the current life cycle. 2) an event that is triggered after navigation is complete, where rootRouter.navigating no longer evaluates to true.
